index.js

```diff
@@ -17,6 +17,30 @@ module.exports = {
     this.shouldPolyfill = emberVersion.lt('3.11.0-alpha.1');
   },
 
+  setupPreprocessorRegistry(type, registry) {
+    if (this.shouldPolyfill) {
+      let pluginObj = this._buildPlugin();
+
+      pluginObj.parallelBabel = {
+        requireFile: __filename,
+        buildUsing: '_buildPlugin',
+        params: {},
+      };
+
+      registry.add('htmlbars-ast-plugin', pluginObj);
+    }
+  },
+
+  _buildPlugin() {
+    return {
+      name: 'component-attributes',
```

Comment:
lol, sorry for late comment, looks like this name should be different

Reply:
hahahaahahah, 😭

Reply:
@rwjblue I'm wondering is any case to use `mut` without combination with `fn`?
like `<MyComponent @onChange={{mut this.content}} />` ? If it's valid case - looks like current transform not cover it

Reply:
This polyfill is **only** for `fn` so I'm not sure it matters what you'd use `{{mut` for if it isn't with `fn` from this libraries perspective.

Also, I'm not sure why you would do `<MyComponent @onChange={{mut this.content}} />`, what is the point of that `mut` invocation from your perspective?

Reply:
`<MyComponent @onChange={{mut this.content}} />` looks pretty interesting from declarative side, mut result assept only one argument, so, onChange must be invoked with only one argument, it's pretty clear.

in case of `{{fn @someExternal}}` we can't imagine how arguments already binded to this function.

`mut` - pretty clear way to get atomic component context properties mutation

Reply:
I think you may be misunderstanding how `{{mut` works, in that position with those arguments `{{mut` does **not** give you a function at all. It gives you the literal value that was in `this.content`. 

`{{mut` only emits a function like thing when passed to "special" helpers (mostly `action`), which uses some private internal details in order to make it work. 

I made a small mockup to show what I mean: https://codesandbox.io/s/my-app-fjvgh

Reply:
is `mut` produce kinda `mutableReference` with `update()` method using with `action`? 
https://github.com/emberjs/ember.js/blob/c21c417705d90837da898059faf93741fa564334/packages/%40ember/-internals/glimmer/lib/helpers/mut.ts

And in end-user-cases it available as  "value" and sometimes as "reference"?

Reply:
Ya, roughly but the reference itself is never exposed to userland (that’s why the fix I had to do in this PR was to wrap in action). 

I personally think the thing you were suggesting that mut return a `{ value, update(){} }` kinda thing would make `mut` **actually** useful and we should consider trying to make a path to that.

Also, FWIW, this roughly equates to the thing I was mentioning around 16:00 in this video https://youtu.be/c0Rl6o9wLX0. 😃

Reply:
Un user-space we able to get `mutable`  as https://ember-twiddle.com/0575c8ca83be668eb694811cd7adb14f?openFiles=templates.components.my-component.hbs%2C